core/src/main/java/com/graphhopper/coll/GHIntArrayList.java

```diff
@@ -0,0 +1,77 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for 
+ *  additional information regarding copyright ownership.
+ * 
+ *  GraphHopper GmbH licenses this file to you under the Apache License, 
+ *  Version 2.0 (the "License"); you may not use this file except in 
+ *  compliance with the License. You may obtain a copy of the License at
+ * 
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.coll;
+
+import com.carrotsearch.hppc.IntArrayList;
+import java.util.Random;
+
+/**
+ *
+ * @author Peter Karich
+ */
+public class GHIntArrayList extends IntArrayList {
+    public GHIntArrayList() {
+        super(10);
+    }
+
+    public GHIntArrayList(int capacity) {
+        super(capacity);
+    }
+
+    public GHIntArrayList(GHIntArrayList list) {
+        super(list);
+    }
+
+    public final GHIntArrayList reverse() {
+        final int[] buffer = this.buffer;
+        for (int start = 0, end = size() - 1; start < end; start++, end--) {
+            // swap the values
+            int tmp = buffer[start];
```

Comment:
Maybe we should define the int outside of the for? Not sure if this gets optimized, but I introduced a "bug" with String once, so I became a bit cautious with these things.